### PR TITLE
Automating Octopus installation

### PR DESCRIPTION
**Pull request by johnsimons**

### CODE REVIEW

File: docs/installation/installing-octopus/automating-installation.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+### 3. Configuration
+
+The MSI installer simply extracts files and adds some shortcuts and event log sources. The actual configuration of Octopus Server is done later, via the script you copied above.
+
+To run the script start an admin shell prompt and paste the script onto it, this should apply all the settings to the new instance.
+
+## Desired State Configuration
+
+Octopus can also be installed via [Desired State Configuration](https://msdn.microsoft.com/en-us/powershell/dsc/overview) (DSC). Using the module from the [OctopusDSC GitHub repository](https://github.com/OctopusDeploy/OctopusDSC).
```
Comment (matt-richardson) [inline docs/installation/installing-octopus/automating-installation.md]: Probably better to reference the package from http://powershellgallery.com/packages/OctopusDSC

**Comment by matt-richardson** [comment]
The image - we've got a docs page about how to do images and highlighting to keep them consistent.

File: docs/installation/installing-octopus/automating-installation.md (hunk 1 above)
Comment (johnsimons) [inline docs/installation/installing-octopus/automating-installation.md]: updated

**Comment by johnsimons** [comment]
@matt-richardson I have updated the highlight according to the rules but given that the button on the left is also green, I am not sure it stands out that well, thoughts ?
show-script

File: docs/installation/installing-octopus/automating-installation.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+### 1. Install MSI on a temporary machine interactively
+In this step we install the MSI on a machine interactively so that we can complete the wizard to add a new instance.
+Follow all the steps in the wizard and once you get to the last screen, click on `Show Script` (**do not click Install**).
+![](show-script.png "width=500")
+
+Copy the script generated onto the clipboard.
+
+Here is an example what the script looks like, **you need to modify this script manually with your settings**:
+```bash
+"C:\Program Files\Octopus Deploy\MyOctopus\Octopus.Server.exe" create-instance --instance "new instance" --config "C:\Octopus\new instance\OctopusServer-new instance.config"
```
Comment (tothegills) [inline docs/installation/installing-octopus/automating-installation.md]: These commands should be for the default installation location and default instance.  Keep the --instance parameter but make it use the default (I forget what it is off the top of my head)

File: docs/installation/installing-octopus/automating-installation.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+The latest Octopus MSI can always be [downloaded from the Octopus Deploy downloads page](https://octopus.com/downloads).
+
+Permalinks to always get the latest MSIs are:
+
+- 32-bit: [https://octopus.com/downloads/latest/WindowsX86/OctopusServer](https://octopus.com/downloads/latest/WindowsX86/OctopusServer)
+- 64-bit: [https://octopus.com/downloads/latest/WindowsX64/OctopusServer](https://octopus.com/downloads/latest/WindowsX64/OctopusServer)
+
+Automating the installation of Octopus Server is a three step process.
+
+### 1. Install MSI on a temporary machine interactively
```
Comment (tothegills) [inline docs/installation/installing-octopus/automating-installation.md]: I don't think installing on a temporary machine is required.  Providing the scripts that does the installation is good enough and consistent with the Tentacle documentation.

File: docs/installation/installing-octopus/automating-installation.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+In this step we install the MSI on a machine interactively so that we can complete the wizard to add a new instance.
+Follow all the steps in the wizard and once you get to the last screen, click on `Show Script` (**do not click Install**).
+![](show-script.png "width=500")
```
Comment (tothegills) [inline docs/installation/installing-octopus/automating-installation.md]: There's a janky border at the top of the image

File: docs/installation/installing-octopus/automating-installation.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        cOctopusServer OctopusServer
+        {
+            Ensure = "Present"
+            State = "Started"
+
+            # Server instance name. Leave it as 'OctopusServer' unless you have more than one instance
+            Name = "OctopusServer"
+
+            # The url that Octopus will listen on
+            WebListenPrefix = "http://localhost:81"
```
Comment (tothegills) [inline docs/installation/installing-octopus/automating-installation.md]: You mention listening on port 80 above, this is port 81

File: docs/installation/installing-octopus/automating-installation.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+In this step we install the MSI on a machine interactively so that we can complete the wizard to add a new instance.
+Follow all the steps in the wizard and once you get to the last screen, click on `Show Script` (**do not click Install**).
+![](show-script.png "width=500")
+
+Copy the script generated onto the clipboard.
+
+Here is an example what the script looks like, **you need to modify this script manually with your settings**:
+```bash
+"C:\Program Files\Octopus Deploy\MyOctopus\Octopus.Server.exe" create-instance --instance "new instance" --config "C:\Octopus\new instance\OctopusServer-new instance.config"
+"C:\Program Files\Octopus Deploy\MyOctopus\Octopus.Server.exe" configure --instance "new instance" --home "C:\Octopus\new instance" --storageConnectionString "Data Source=(local)\SQLEXPRESS;Initial Catalog=foo;Integrated Security=True" --upgradeCheck "True" --upgradeCheckWithStatistics "True" --webAuthenticationMode "UsernamePassword" --webForceSSL "False" --webListenPrefixes "http://localhost:80/" --commsListenPort "10943" --serverNodeName "JOHNSBOX"
```
Comment (tothegills) [inline docs/installation/installing-octopus/automating-installation.md]: The configure command can be split into multiple commands to make it more legible

File: docs/installation/installing-octopus/automating-installation.md (hunk 4 above)
Comment (johnsimons) [inline docs/installation/installing-octopus/automating-installation.md]: fixed

File: docs/installation/installing-octopus/automating-installation.md (hunk 5 above)
Comment (johnsimons) [inline docs/installation/installing-octopus/automating-installation.md]: fixed

File: docs/installation/installing-octopus/automating-installation.md (hunk 6 above)
Comment (johnsimons) [inline docs/installation/installing-octopus/automating-installation.md]: done the split

File: docs/installation/installing-octopus/automating-installation.md (hunk 2 above)
Comment (johnsimons) [inline docs/installation/installing-octopus/automating-installation.md]: made it placeholders like `<instance name>`

File: docs/installation/installing-octopus/automating-installation.md (hunk 3 above)
Comment (johnsimons) [inline docs/installation/installing-octopus/automating-installation.md]: After discussing with @tothegills we have decided to leave it this way because of the different permutations of configuration, would make a very long and hard to follow script

**Comment by johnsimons** [comment]
@tothegills changes requested done, can u check it out again
